conf/report/report-template.html

```diff
@@ -33,9 +38,250 @@
           {% for tool in report %}
           { "orderDataType": "test-status" },
           {% endfor %}
-        ]
+        ],
+        initComplete: function() {
+          $("#report_table thead tr th a").each(function(){
+            toolnames.push(this.innerHTML);
+          });
+        }
       });
+      tool_array = table.columns()[0];
+      tool_array = tool_array.splice(2);
+      col_array = table.columns()[0];
     } );
+
+    var iter;
+    var tool_selected = false;
+    
+    const filter_types = {
+      coverage: {
+        name: "Coverage",
+        operators: ["<", "<=", ">", ">=", "="],
+        value_field_factory: function(id, span_id=0) {
+          span_id = parseInt(span_id)
+          if ($(`li[data-uid=${id}] span[data-uid=${span_id+1}]`).length > 0)
+            $(`li[data-uid=${id}] span[data-uid=${span_id+1}]`).remove()
+          var iter = $(`li[data-uid=${id}] span`).length
+          if (iter === 0)
+            var filter_condition = $(`<span data-uid=${iter}></span>`).insertAfter(`li[data-uid=${id}] .filter-entry-type`)
+          else
+            var filter_condition = $(`<span data-uid=${iter}></span>`).insertAfter(`li[data-uid=${id}] span[data-uid=${iter-1}]`)
+          var filter_entry_operator =  $('<select class="filter-entry-operator"><option value=""></option></select>').appendTo(`li[data-uid=${id}] span[data-uid=${iter}]`)
+          var filter_entry_value =  $('<select class="filter-entry-value"><option value=""></option></select>').appendTo(`li[data-uid=${id}] span[data-uid=${iter}]`)
```

Comment:
this seemed to repeat across the `tool` and `type` filters too in the `value_field_factory`. would it make sense to write a function to remove the repetition?

Reply:
Sure